pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -200,24 +244,82 @@ func (d *PmemCSIDriver) reconcileDeploymentChanges(r *ReconcileDeployment, chang
 		}
 	}
 
+	d.Annotations[VersionAnnotation] = r.operatorVersion
+
 	return false, nil
 }
 
-func (d *PmemCSIDriver) deployObjects(r *ReconcileDeployment) error {
-	objects, err := d.getDeploymentObjects()
-	if err != nil {
-		return err
+func (d *PmemCSIDriver) deleteObsoleteObjects(r *ReconcileDeployment, newObjects []apiruntime.Object) error {
+	type NameGVK struct {
+		name      string
+		namespace string
+		gvk       schema.GroupVersionKind
 	}
-	for _, obj := range objects {
-		if err := r.Create(obj); err != nil {
+
+	foundList := []NameGVK{}
+
+	for _, gvk := range AllObjectTypes {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(gvk)
+		opts := &client.ListOptions{
+			Namespace: d.namespace,
+		}
+
+		klog.Infof("Fetching '%s' list with options: %v", gvk, opts.Namespace)
+		if err := r.client.List(context.TODO(), list, opts); err != nil {
 			return err
 		}
+
+		for _, obj := range list.Items {
+			for _, owner := range obj.GetOwnerReferences() {
+				if owner.UID == d.GetUID() {
+					foundList = append(foundList, NameGVK{
```

Comment:
This collection of found objects in a list is unnecessary. The code becomes simpler (less variables, smaller functions, function names that explain what the code does) when moving loop bodies below into functions (`objectIsObsolete`, `checkExistingObject`). The you just call `checkExistingObject` here.

Reply:
Removed the found list.